### ISSUE TITLE
Implement getStarred2

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#1838]
REST considerations. Related airsonic/airsonic#1770, #1752.
Prerequisites: [v111.6.0](https://github.com/tesshucom/jpsonic/milestone/62)

It seems to be one of the things not implemented in Legacy. I don't remember if it was fixed in Jpsonic.
There is some ambiguity in ID3 handling, which should be fixed in v111.6.0 or later. (Or they may be fixed at the same time.)

**Comment by masterblaster79** [comment]
My observation is the following (tested using webui and dsub android client):
- Starring tracks is fine in any case, they are stored in the "starred_media_file" table.
- Starring artists from webui/dsub(folder based) stores the artist folder in "starred_media_file" and is not using the "starred_artist" table
- Starring artist from dsub(tag based), it's stored in the "starred_artist" table
- Starring albums from webui/dsub(folder based) stores the album folder in "starred_media_file" and is not using the "starred_album" table
- Starring albums from API when organizing by tags instead of folders it's stored in the "starred_album" table

So, starring albums/artists is inconsistent whether tag based or folder based browsing is being used.

**Comment by tesshucom** [comment]
As I remember. . . In the Subsonic API, I think that File Structure and ID3 were defined to use different IDs. (In that sense, I think it was a specification that should be looked at first among many API method specifications.)

The API specification does not define whether tag-based starring and folder-based starring should be correlated. Therefore, I think the implementation specifications are up to each product. I think that Airsonic could not keep up with maintenance on the ID3 side. (Although the definition of the database borrowed a lot from Subsonic.) 

What I'm trying to fix is to repair the functionality using the existing table first. Therefore, I think that it is a correction about what you just pointed out.

It's not that difficult, but if e start with something easy, we'll run out of resources. Because there are many similar easy fixes. Core bugs won't be fixed if you get stuck there. That's the wrong path many media servers take. Therefore, although it is decided that it will be fixed, please understand that the timing may be just not now.

The current subject is scan fixes. However, this does not mean that no other amendments will be accepted during that period. If you create a pull request, it would be most welcome.

**Comment by tesshucom** [comment]
This is under investigation and additional testing is planned. At this time, it has been concluded that there is no problem with the Airsonic controller. From Subsonic to Airsonic, there are no flaws in the design and implementation of the Star-related implements on Controller Layer. 

 - airsonic/airsonic#1770 etc

To summarize, this is a misunderstanding of the user's specifications. The FileStructure Star and the Id3 Star are different systems of stars and they are never orthogonal. (Any designer with a certain level of design skill would have a similar opinion...) 

 > This would be possible if you could set absolute hierarchical rules in FileStructure and have users follow them strictly. But that's a very unrealistic assumption. Subsonic does not equate a FileStructure structure with an Id3 structure.

So, in general, there is nothing wrong with Subsonic/Airsonic's approach to Star, and that will not change for Jpsonic. 

___

However, there were significant issues with persistence of that data. Airsonic has sometime had problems with 'duplicate entries' in the data.  

To solve this problem, Jpsonic employs the following strategies:

 - Subsonic/Airsonic's Dao has a fatal method called createOrUpdate. This is one of the factors that creates ghost data. It is very difficult to reproduce, but depending on the timing, there may be conflicts between different functions. In Jpsonic, Create and Update are separated. Obviously, all CRUD operations must have clearly stated conditions and Business logic level transaction concurrency must be ensured.
 - Subsonic and Airsonic do not have the ability to self-clean when ghost data occurs. Therefore, in Jpsonic, if IgnoreTimestamp is enabled, the Artist and Album in ID3 will be All deleted and inserted. 
   - While this implementation was maintained, all of the processes that could result in ghost data were fixed. (As you can imagine, this fix is ​​very time consuming.)

The downside of this approach is that if you run a full scan with IgnoreTimestamp, the Star with Id3 in the external reference will be removed. Such a policy is adopted for the following reasons.

 - From a user perspective, the removal of stars is a terrible feature and frankly seems like a bug. That's true, but data duplication is a much bigger nightmare. This could develop into a very obscure and unknown problem, and it is difficult to predict the extent of the impact. If we were asked to fix both, given limited resources, you would probably have to accept the former bug in the interim.
 - There are fewer Subsonic apps actively adopting Id3 Star than there are apps adopting FileStructure Star. Of course, this is a matter of degree. If Id3 Star experiences some temporary functionality restrictions, it doesn't mean that all users will be affected. Data CRUD bugs as Duplicates affect all users. It affects all functionality... and developing new features is completely pointless as long as this bug exists.

___

Starting with v114.2, the use of Jpsonic's IgnoreTimestamp scan will change (revert back to original). It's been a long effort, but it will ensure that Star works as designed, and it will solve the pesky problem of data duplication.
